figures/scripts/fig5_figS6-immune-deconv-panels.R

```diff
@@ -0,0 +1,377 @@
+                               "Medulloblastoma",
+                               "Schwannoma",
+                               "Neurofibroma Plexiform",
+                               "Other embryonal tumor",
+                               "Atypical Teratoid Rhabdoid Tumor",
+                               "Meningioma",
+                               "Dysembryoplastic neuroepithelial tumor")
+
+## Input and output files -------------------------------------
+
+hist_file <- file.path(data_dir, "pbta-histologies.tsv")
+palette_file <- file.path(palette_dir, "broad_histology_cancer_group_palette.tsv")
+quantiseq_file <- file.path(analysis_dir, "quantiseq_deconv-output.rds")
+polya_file <- file.path(data_dir, "pbta-gene-expression-rsem-fpkm-collapsed.polya.rds")
+stranded_file <- file.path(data_dir, "pbta-gene-expression-rsem-fpkm-collapsed.stranded.rds")
+
+# Figure 5 output:
+cancer_group_pdf <- file.path(output_dir_fig5, "quantiseq-cell_types-cancer_groups.pdf")
+cd274_mb_plot_pdf <- file.path(output_dir_fig5, "cd274_expression_mb_subtypes.pdf")
+
+# Figure S6 output:
+subtypes_pdf <- file.path(output_dir_figS6, "quantiseq-cell_types-molecular_subtypes.pdf")
+cd8_cd4_ratio_pdf <- file.path(output_dir_figS6, "cd8_cd4_ratio.pdf")
+
+
+# Prepare data for plots --------------
+
+# Read in clinical data and palette file
+histologies_df <- read_tsv(hist_file, guess_max = 10000)
+palette_df <- readr::read_tsv(palette_file)
+
+# Read in analysis data
+quantiseq <- read_rds(quantiseq_file)
+polya <- read_rds(polya_file)
+stranded <- read_rds(stranded_file)
+
+# Prepare palette/histology data data
+palette_mapping_df <- histologies_df %>%
+  # RNA-Seq samples only, *****BOTH****** polya and stranded
+  filter(experimental_strategy == "RNA-Seq") %>%
+  # Identifiers
+  select(Kids_First_Biospecimen_ID,
+         Kids_First_Participant_ID,
+         sample_id,
+         broad_histology,
+         cancer_group, 
+         molecular_subtype) %>%
+  # Add in hex codes & display grouping
+  left_join(palette_df,
+            by = c("broad_histology", "cancer_group")) %>%
+  select(Kids_First_Biospecimen_ID,
+         contains("broad_histology"), 
+         contains("cancer_group"),
+         molecular_subtype)
+
+
+#### Prepare quanTIseq data --> `quantiseq_subset`
+# spread and create cd8/cd4 ratio
+quantiseq_spread <- quantiseq %>%
+  # get a column for each cell type
+  spread(cell_type, score) %>%
+  mutate(cd8_cd4_ratio = `T cell CD8+` / `T cell CD4+ (non-regulatory)`)
+
+quantiseq_gather <- quantiseq_spread %>%  
+  gather(cell_type, score, -c(sample, library, method))
+
+# First, find the molecular subtypes in the of interest cancer groups AND excluding unclassified, with >=3 samples
+subtypes_of_interest <- palette_mapping_df %>%
+  filter(cancer_group_display %in% cancer_groups_of_interest, 
+         !(str_detect(molecular_subtype, "To be classified"))) %>%
+  count(molecular_subtype) %>%
+  filter(n >= 3) %>%
+  pull(molecular_subtype)
```

Comment:
The one thing I noticed over in the manuscript was that S6 immune deconv panels have the HGG subtype removed, I thought because other HGG was no longer in the plots. But, I see they did not return in the plots - I am pretty sure that HGG, H3 WT has N>3 - would you mind checking that here - we probably should recover them with updating these figures.

Reply:
Ok, I see what's happening - I had defined `cancer_groups_of_interest` for *panel 5C* on line 36 as  - 
```
cancer_groups_of_interest <- c("Pilocytic astrocytoma", 
                               "Diffuse midline glioma",
                               "Craniopharyngioma", 
                               "Ganglioglioma",
                               "Ependymoma", 
                               "Medulloblastoma",
                               "Schwannoma",
                               "Neurofibroma Plexiform",
                               "Other embryonal tumor",
                               "Atypical Teratoid Rhabdoid Tumor",
                               "Meningioma",
                               "Dysembryoplastic neuroepithelial tumor")
```

This code you pointed out filters to those cancer groups, and "other HGG" is not in that list. So, for the S6 plots, we probably don't want to use this filtering. If I don't filter to that list and include all cancer groups, here are _all_ the molecular subtypes with N>= 3, arranged by subtype. We definitely can't be plotting all 33 of these though in one plot, so we will need to discuss a different approach to filtering down I think?

```
1 CHDM, poorly differentiated          4
 2 CNS Embryonal, NOS                  14
 3 CNS NB-FOXR2                         3
 4 CRANIO, ADAM                        24
 5 DMG, H3 K28                         20
 6 DMG, H3 K28, TP53 activated          9
 7 DMG, H3 K28, TP53 loss              39
 8 EPN, ST RELA                        28
 9 ETMR, C19MC-altered                  5
10 EWS                                 11
11 GNG, BRAF V600E                     12
12 GNG, KIAA1549-BRAF                   5
13 GNG, other MAPK                      4
14 GNG, RTK                             3
15 GNG, wildtype                       14
16 HGG, H3 G35                          4
17 HGG, H3 wildtype                    37
18 HGG, H3 wildtype, TP53 activated     5
19 HGG, H3 wildtype, TP53 loss         20
20 LGG, BRAF V600E                     26
21 LGG, BRAF V600E, CDKN2A/B            5
22 LGG, FGFR                            8
23 LGG, IDH                             3
24 LGG, KIAA1549-BRAF                 113
25 LGG, NF1-germline                    6
26 LGG, other MAPK                     12
27 LGG, RTK                            10
28 LGG, wildtype                       34
29 MB, Group3                          14
30 MB, Group4                          50
31 MB, SHH                             31
32 MB, WNT                             10
33 SEGA, wildtype                      11
```

Reply:
oh wow, can you do a quick cd8_cd4_ratio.pdf plot of those - not ideal, but we can select based on what might have higher ratios, if more than cranio + MB group 3/4? If nothing else, then we can assess? For the larger subtype plot, the major thing I highlight in the paper is CD8 for MB group 3/4 and monocytes, myeloid DCs for DMGs. 

Also, it looks like the number and order of subtypes in`quantiseq-cell_types-molecular_subtypes.pdf` and `cd8_cd4_ratio.pdf` are slightly different with the EPN, extra embryonals in the first and cranio being in a different spot. The struggle.

Reply:
Yup, here's that plot with all N>=3 subtypes across all cancer groups. One HGG, H3 WT sample is a bit elevated, but nothing else really stands out strongly.
[cd8_cd4_ratio_ALL-SUBTYPES.pdf](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/9193615/cd8_cd4_ratio_ALL-SUBTYPES.pdf)


> Also, it looks like the number and order of subtypes inquantiseq-cell_types-molecular_subtypes.pdf and cd8_cd4_ratio.pdf are slightly different with the EPN, extra embryonals in the first and cranio being in a different spot. The struggle.

The struggle indeed 😂 Since these plots are next to each other, I think it would make sense to show the same molecular subtypes in each of them, and in the same order. Maybe let's pick for CD8/CD4 and then use that in `quantiseq-cell_types-molecular_subtypes.pdf`?

Reply:
ok, how about cranio, dmg, hgg, MB, and the embryonal types (since we have MB) here for both plots ... or if easier, just remove GNG/LGG/SEGA bc there is nothing really to note in those subtypes, and keep the rest (would then also include EPN)?